src/server/scripts/EasternKingdoms/SunkenTemple/instance_sunken_temple.cpp

```diff
@@ -189,20 +249,80 @@ class instance_sunken_temple : public InstanceMapScript
             go->SummonCreature(NPC_ATALALARION, atalalarianPos, TEMPSUMMON_CORPSE_TIMED_DESPAWN, 10min);
         }
 
+        bool SetBossState(uint32 type, EncounterState state) override
+        {
+            if (!InstanceScript::SetBossState(type, state))
+                return false;
+            switch (type)
+            {
+            case DATA_JAMMAL_AN_THE_PROPHET:
+                if (state == DONE)
+                {
+                    if (Creature* creature = instance->GetCreature(ShadeOfEranikusGUID))
+                    {
+                        creature->RemoveUnitFlag(UNIT_FLAG_IMMUNE_TO_PC);
+                        creature->RemoveUnitFlag(UNIT_FLAG_IMMUNE_TO_NPC);
+                    }
+                }
+                SaveToDB();
+                break;
+            default:
+                break;
+            }
+            return true;
+        }
+
          void SetData(uint32 type, uint32 data) override
          {
-            if (type == EVENT_STATE)
+            switch (type)
+            {
+            case EVENT_STATE:
                 State = data;
+                break;
+            case DATA_ELITE_TROLLS:
+                EliteTrollsKilled = data;
+                if (EliteTrollsKilled == 6)
+                {
+                    //instance->LoadGrid(-425.89f, -86.07f);    
+                    if (Creature* jammal = instance->GetCreature(JammalAnTheProphetGUID))
+                        jammal->RemoveUnitFlag(UNIT_FLAG_IMMUNE_TO_PC);
+                    /*if (GameObject* forceField = instance->GetGameObject(ForceFieldGUID))
+                        forceField->SetGoState(GO_STATE_ACTIVE);*/
```

Comment:
**circleci: codestyle**: 1 line containing whitespace at the end:
```cpp
                    //instance->LoadGrid(-425.89f, -86.07f);    
```
Other than that, it is recommended not to add comment lines for anything except comments. The code lines you have added, but commented out, need to be removed or uncommented if you are planning to use them at all: